### PR TITLE
Tpetra: use GIDs to convert from point matrix to block matrix

### PR DESCRIPTION
**Pull request by kliegeois**
@trilinos/tpetra 
@jhux2 

This PR fixes the pts->block conversion by using GIDs instead of relying on local IDs.

### CODE REVIEW

**Comment by achauphan** [comment]
> Status Flag 'Pull Request AutoTester' - Failure: Job Trilinos_PR_gcc-8.3.0 on Jenkins server at https://do.sandia.gov/trilinos-ci Does Not Exist
> 
>     * Other jobs have been previously started - We must stop them...

AutoTester is currently broken due to some issues with our Jenkins services. I've made and pinned a GitHub issue about this and will update and unpinned it when the issue has resolved. 

- #12915

**Comment by kliegeois** [comment]
Thanks @jhux2 for fixing the dashboard error!

**Review by jhux2 — APPROVED** [review]
This looks fine to me.  I have one minor comment about a variable name, but it's not enough to block this.

File: packages/tpetra/core/src/Tpetra_BlockCrsMatrix_Helpers_def.hpp [hunk 1]
```diff
@@ -362,9 +366,22 @@ namespace Tpetra {
           blockRowptr(i) = offset_b;
 
           const LO offset_p = pointRowptr(i*blockSize);
+          const LO offset_p_max = pointRowptr((i+1)*blockSize);
+
+          LO filled_block = 0;
+          for (LO p_i=0; p_i<offset_p_max-offset_p; ++p_i) {
+            auto bcol_GID = localPointColMap.getGlobalElement(pointColind(offset_p + p_i))/blockSize;
+            auto lcol_GID = localMeshColMap.getLocalElement(bcol_GID);
```
[the remark this answers is not in the document]
Comment (kliegeois) [reply packages/tpetra/core/src/Tpetra_BlockCrsMatrix_Helpers_def.hpp:374]: Yes, it was supposed to be `bcol_LID` good catch!

**Comment by kliegeois** [comment]
Thanks for the review @jhux2 !

**Pull request merged**